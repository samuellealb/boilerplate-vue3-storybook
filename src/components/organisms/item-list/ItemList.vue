<template>
  <div class="item-list">
    <template v-if="hasItems">
      <item-card v-for="item in items" :item="item" :key="item.id" @delete-item="onDelete" />
    </template>
    <template v-else>
      <h2>No items loaded =/</h2>
    </template>
  </div>
</template>

<script>
import ItemCard from '@/components/molecules/item-card/ItemCard.vue'
import './item-list.css'
import { computed } from 'vue';

export default {
  name: 'ItemList',
  components: {
    ItemCard
  },
  props: {
    items: {
      type: Array,
      default: () => [],
      required: true
    }
  },
  emits: ['delete-item'],
  setup(props, { emit }) {
    function onDelete (id) {
      emit('delete-item', id);
    }
    const hasItems = computed(() => props.items.length > 0)
    return { onDelete, hasItems }
  },
}
</script>