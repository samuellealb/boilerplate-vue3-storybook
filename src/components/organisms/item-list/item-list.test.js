import { mount } from '@vue/test-utils'
import { test, expect } from 'vitest'
import ItemList from './ItemList.vue'

test('mount component', async () => {
  expect(ItemList).toBeTruthy()

  const defaultItemList = mount(ItemList, {
    props: {
      items: [
        {
          id: '1',
          name: 'Testing Item List 1'
        },
        {
          id: '2',
          name: 'Testing Item List 2'
        },
        {
          id: '3',
          name: 'Testing Item List 3'
        },
      ]
    }
  })
  expect(defaultItemList.html()).toMatchSnapshot()

  const noItemList = mount(ItemList, {
    props: {
      items: []
    }
  })
  expect(noItemList.html()).toMatchSnapshot()
})

test('Click on second button emit "delete-item" with item id value', async() => {
  const wrapper = mount(ItemList, {
    props: {
      items: [
        {
          id: 55,
          name: 'Testing Item List 1'
        },
        {
          id: 235,
          name: 'Testing Item List 2'
        },
        {
          id: 112,
          name: 'Testing Item List 3'
        },
      ]   
    }
  })

  wrapper.find('.item-card:nth-child(2) button').trigger('click')
  await wrapper.vm.$nextTick()
  expect(wrapper.emitted('delete-item')[0]).toEqual([235])
})