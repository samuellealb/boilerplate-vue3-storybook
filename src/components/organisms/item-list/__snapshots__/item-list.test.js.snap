// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"item-list\\">
  <div class=\\"item-card\\">
    <div class=\\"item-card__title\\">Testing Item List 1</div>
    <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
  </div>
  <div class=\\"item-card\\">
    <div class=\\"item-card__title\\">Testing Item List 2</div>
    <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
  </div>
  <div class=\\"item-card\\">
    <div class=\\"item-card__title\\">Testing Item List 3</div>
    <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
  </div>
</div>"
`;

exports[`mount component 2`] = `
"<div class=\\"item-list\\">
  <h2>No items loaded =/</h2>
</div>"
`;
